src/openfunction/context.ts

```diff
@@ -196,3 +212,22 @@ export enum TraceProviderType {
    */
   OpenTelemetry = 'opentelemetry',
 }
+
+/**
+ * The specification of the component
+ * @public
+ */
+export interface ComponentSpec {
```

Comment:
[suggestion] We could follow [v1beta1 spec](https://github.com/OpenFunction/OpenFunction/blob/main/pkg/core/serving/common/functioncontext.go#L21) to define `OpenFunctionContextV1Beta1` and make it union with `OpenFunctionContext`.